File: NeuralNet/SnakeAdapter.m
//
//  SnakeAdapter.m
//  SnakeNet
//
//  Created by Maya Luna Celeste on 4/30/20.
//  Copyright © 2020 Maya Luna Celeste. All rights reserved.
//

#import "SnakeAdapter.h"

@implementation SnakeAdapter

- (SnakeInputType)snakeInput{
    int value = 0;
    for(int i = 0; i < [self.outputs count]; i++){
        if([[self.outputs objectAtIndex:i] doubleValue] > [[self.outputs objectAtIndex:value] doubleValue]) {
            value = i;
        }
    }
    
    SnakeInputType input = SnakeInputTypeError;
    
    if (value == 0){
        input = SnakeInputTypeUp;
    } else if (value == 1){
        input = SnakeInputTypeDown;
    } else if (value == 2){
        input = SnakeInputTypeLeft;
    } else if (value == 3){
        input = SnakeInputTypeRight;
    }
    return input;
}

@end

File: NeuralNet/SnakeAdapter.swift
//
//  SnakeAdapter.swift
//  NeuralNet
//
//  Created by Maya Luna Celeste on 8/5/20.
//  Copyright © 2020 Maya Luna Celeste. All rights reserved.
//

import Cocoa

class SnakeAdapter: NSObject {
    
    var netOuts : [Double] = [0,0,0,0]
    
    enum SnakeMove : Int {
        case up = 0
        case down = 1
        case left = 2
        case right = 3
        case error = 4
    }
    
    init(network : Network) {
        netOuts = network.outs
    }
    
    func moveType() -> SnakeMove {
        if netOuts.count != 4 { return SnakeMove.error }
        var val : Int = 0
        for i in 0 ..< netOuts.count { if netOuts[i] > netOuts[val] { val = i } }
        return SnakeMove(rawValue: val) ?? SnakeMove.error
    }
    
}
